driver/bpf/fillers.h

```diff
@@ -89,6 +89,49 @@ static __always_inline int bpf_##x(void *ctx)				\
 									\
 static __always_inline int __bpf_##x(struct filler_data *data)		\
 
+static __always_inline struct inode *get_file_inode(struct file *file)
+{
+	if (file) {
+		return _READ(file->f_inode);
+	}
+	return NULL;
+}
+
+static __always_inline enum ppm_overlay get_overlay_layer(struct file *file)
+{
+	if (!file)
+	{
+		return PPM_NOT_OVERLAY_FS;
+	}
+	struct dentry* dentry = NULL;
+	bpf_probe_read_kernel(&dentry, sizeof(dentry), &file->f_path.dentry);
+	struct super_block* sb = (struct super_block*)_READ(dentry->d_sb);
+	unsigned long sb_magic = _READ(sb->s_magic);
+
+	if(sb_magic != PPM_OVERLAYFS_SUPER_MAGIC)
+	{
+		return PPM_NOT_OVERLAY_FS;
+	}
+
+	char *vfs_inode = (char *)_READ(dentry->d_inode);
+	struct dentry *upper_dentry = NULL;
+	bpf_probe_read_kernel(&upper_dentry, sizeof(upper_dentry), (char *)vfs_inode + sizeof(struct inode));
+	if(!upper_dentry)
+	{
+		return PPM_OVERLAY_LOWER;
```

Comment:
Not 100% sure but I don't think that `!upper_dentry` -> `PPM_OVERLAY_LOWER` WDYT?

```suggestion
		return PPM_NOT_OVERLAY_FS;
```